# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# DO NOT GRADE THIS GROUP LAB - OTHER GROUP MEMBER SUBMITTED

# STAT 207 Lab Assignment 11 - Group Part  - [13.25 Points]



## <u>Case Study</u>: Predicting Airbnb Superhost Status

Suppose that you have a spare room in your house (in Seattle) and you are considering advertising this room to guests on Airbnb. Airbnb superhosts are considered to be experienced hosts who provide a shining example for other hosts, and extraordinary experiences for their guests. Once a host reaches Superhost status, a superhost badge will automatically appear on their listing and profile to help guests identify them. You would like to assess your chances of being named a superhost with your property.

The following dataset is a sample (assume random) of available Airbnb listings in Seattle, WA. These listings were collected in January 2016, and filtered to just contain listings from the five most popular Seattlbe neighborhoods (for Airbnb listings) and just contain listings that are either in a house or apartment property.

This dataset contains the following variables.

**Listing Information**
The dataset contains the following information about the Airbnb *listing*:
* <u>price</u>: price of the listing (per night)(in US dollars)
* <u>review_scores_rating</u>: the average rating of the listing [0,100] (100 is the best)
* <u>number_of_reviews</u>: the number of reviews for the listing
* <u>security_deposit</u>: the security deposit required for the listing (in US dollars)
* <u>cleaning_fee</u>: the cleaning fee required for the listing (in US dollars)
* <u>neighborhood</u>: the neighborhood of Seattle the listing is located in
* <u>property_type</u>: is the listing in a 'House' or 'Apartment'
* <u>room_type</u>: is the listing a 'Entire home/apt', 'Private room', or 'Shared room'
* <u>accommodates</u>: how many guests will the listing accommodate
* <u>bathrooms</u>:how many bathrooms does the listing have
* <u>beds</u>: how many beds does the listing have

**Host Information**
The dataset also contains the following information about the *host* of the given Airbnb listing:
* <u>host_is_superhost</u>: is the host a "superhost": t=True, f=False
* <u>host_has_profile_pic</u>: does the host have a profile pic in their bio: t=True, f=False
* <u>host_response_time</u>: how fast will the host respond to requests (on average)

## Research Goals

The primary research goal that we will pursue in this group lab will be to build a predictive model that effectively predicts the `probability a Seattle Airbnb host is a **superhost**` for ***LISTINGS IN THIS DATASET*** given some combination of the available explanatory variables.



## Points

<p>&nbsp;</p>
<table style="border: none;border-collapse: collapse;width:102pt;">
    <tbody>
        <tr>
            <td style="color:black;font-size:15px;font-weight:700;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;width:51pt;">Problem</td>
            <td style="color:black;font-size:15px;font-weight:700;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:general;vertical-align:bottom;border:.5pt solid windowtext;border-left:none;width:51pt;">Points</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">1.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">0.25</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">1.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">0.5</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">2.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">0.5</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">2.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">0.75</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1.25</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.3.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.3.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.3.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.4.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.4.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.4.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.4.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.4.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;height:14.25pt;border-top:none;">3.4.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid windowtext;border-top:none;border-left:none;">1</td>
        </tr>
    </tbody>
</table>

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.model_selection import train_test_split

import statsmodels.formula.api as smf

from sklearn.linear_model import Ridge, Lasso, ElasticNet, LinearRegression
from sklearn.metrics import r2_score

from sklearn.preprocessing import StandardScaler

import warnings
from sklearn.exceptions import ConvergenceWarning

# Ignore scikit-learn ConvergenceWarnings
warnings.filterwarnings("ignore", category=ConvergenceWarning)
warnings.filterwarnings("ignore", category=UserWarning)

## 1. Basic Data Examination and Manipulation

### 1.1. Basics
1. Read the `seattle_airbnb_listings.csv` file into a dataframe.

2. Display your dataframe.

3. Show how many rows your dataframe has.



In [2]:
df = pd.read_csv("seattle_airbnb_listings.csv")
df.head()

price  review_scores_rating  number_of_reviews  security_deposit  \
0    300                   100                 24               500   
1    149                    96                 11               300   
2     95                    95                 79               150   
3    105                   100                 13               500   
4    140                    99                 30               250   

   cleaning_fee neighborhood property_type        room_type  accommodates  \
0            95  Wallingford         House  Entire home/apt             5   
1           105  Wallingford     Apartment  Entire home/apt             6   
2            40  Wallingford     Apartment  Entire home/apt             3   
3            50  Wallingford         House     Private room             2   
4            65  Wallingford         House  Entire home/apt             2   

   bathrooms  beds  host_response_time  host_response_rate host_is_superhost  \
0        1.5     3  within a few hours                 1.0                 t   
1        1.0     3      within an hour                 1.0                 f   
2        1.0     2      within an hour                 1.0                 f   
3        2.0     1  within a few hours                 1.0                 t   
4        1.0     1      within an hour                 1.0                 t   

  host_has_profile_pic host_identity_verified  
0                    t                      t  
1                    t                      t  
2                    t                      t  
3                    t                      t  
4                    t                      t

In [3]:
shape1 = df.shape[0]
print(f"The dataframe has {shape1} rows")

The dataframe has 325 rows


### 1.2. 0/1 Response Variable

Create a new 0/1 variable in your dataset in which `superhosts=1` and `nonsuperhosts=0`. 

In [4]:
df["host_is_superhost"] = df["host_is_superhost"].replace({"t": 1, "f": 0})

## 2. Model

In the interest of time, let's build a logistic regression model, which will predict the `probability that a host is a superhost` given:

* `review_scores_rating`
* `property_type`
* `room_type`

### 2.1. Fit the Model

Fit this "full" logistic regression model and show the summary output table.

In [5]:
model = smf.logit(formula="host_is_superhost~review_scores_rating+property_type+room_type",data=df).fit()
model.summary()

Optimization terminated successfully.
         Current function value: 0.524382
         Iterations 8


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  325
Model:                          Logit   Df Residuals:                      320
Method:                           MLE   Df Model:                            4
Date:                Wed, 03 Apr 2024   Pseudo R-squ.:                  0.1360
Time:                        10:14:38   Log-Likelihood:                -170.42
converged:                       True   LL-Null:                       -197.24
Covariance Type:            nonrobust   LLR p-value:                 6.264e-11
=============================================================================================
                                coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------------
Intercept                   -21.5917      4.158     -5.192      0.000     -29.742     -13.442
property_type[T.House]       -0.1108      0.293     -0.378      0.705      -0.685       0.463
room_type[T.Private room]     0.0420      0.370      0.113      0.910      -0.683       0.767
room_type[T.Shared room]     -2.1820      1.059     -2.059      0.039      -4.259      -0.105
review_scores_rating          0.2173      0.043      5.056      0.000       0.133       0.302
=============================================================================================
"""

### 2.2. Model Equation

Write the equation for the resulting logistic regression model that has the predicted **odds** on the left-hand side. Make sure to use the appropriate notation discussed in class.

$\widehat{odds} = \dfrac{
\widehat{p}
}{
1 - \widehat{p}
} = e^{\begin{aligned}
    &- 21.5917 \\
    &- 0.1108*property\_type[T.House]\\
    &+ 0.0420*room\_type[T.Private room]\\
    &- 2.1820*room\_type[T.Shared room]\\
    &+ 0.2173*review\_scores\_rating\\
    \end{aligned}
}
$

## 3. Slope and Intercept Interpretations

### 3.1. Baseline Odds

Calculate the baseline odds for this model and put it into words. Make sure to use non-misleading language discussed in class.

In [6]:
baseline_odds = np.exp(-21.5917)

4.196080729325872e-10

We expect the odds of a listing with a property type of Apartment, a room type of Entire Home/Apt, and a review score rating of 0, to be a superhost to be 4.196e-10 to 1

### 3.2. Prose Baseline Odds

The baseline odds that you calculated in 3.1 is a numerical odds. Put this numerical odds into prose form.

In [7]:
1/4.196e-10

2383222116.3012395

The prose odds are 1 to 2383222116.

### 3.3. Odds Multiplier for Review Scores Rating

#### 3.3.1. Odds Multiplier

Calculate the odds multiplier for review scores rating and put it into words. Make sure to use non-misleading language discussed in class.

In [13]:
np.exp(0.2173)

1.242716861279481

All other variables held equal, if we were to increase the review score by 1, we would expect, on average, for the odds to increase by a factor of 1.2427.

#### 3.3.2. Applying the Odds Multiplier

Suppose that an Airbnb host currently has a review scores rating of 8. The odds of her being a superhost was predicted to be 2 by this model. If she were to increase her review scores rating to 9, what would the model predict her new odds to be?

In [14]:
2 * 1.2427

2.4854

#### 3.3.3. Review Scores and Superhost Status

Visualize the relationships between `host_is_superhost` and `review_scores_rating`.

Does your odds multiplier of `review_scores_rating` interpretation match what your visualization suggests about the relationship between `host_is_superhost` and `review_scores_rating`? Explain.

### 3.4. Odds Ratio of House vs. Apartment Listings

#### 3.4.1. Odds Ratio

Calculate the odds ratio for <u>house</u> vs. <u>apartment</u> listings in this model and put it into words. Make sure to use non-misleading language discussed in class.

In [16]:
np.exp(0.1108)

1.1171714502194958

All other variables held equal, we expect, on average, that the odds of a listing with a House to be 1.117 times higher than the odds of a listing with an Apartment.

#### 3.4.2. Applying the Odds Ratio

Suppose that an Airbnb host is currently listing a private room in their <u>apartment</u>. The model predicts the odds that he is a superhost to be 0.6. If he were to also list a private room in a <u>house</u>, what would the model predict the odds of him being a superhost to be?

#### 3.4.3. Property Type and Superhost Status

Visualize the relationships between `host_is_superhost` and `property_type`. 

Does your odds ratio interpretation from 3.4.1 match what your visualization suggests about the relationship between `host_is_superhost` and `property_type`? Explain.

*(Hint: `property_type` should be on your x-axis in this visualization in order to best answer the question above).*

### 3.5. Odds Ratio of Shared Room vs. Whole House/Apartment Listings

#### 3.5.1. Odds Ratio

Calculate the odds ratio for <u>shared room</u> vs. <u>whole house/apartment</u> listings in this model and put it into words. Make sure to use non-misleading language discussed in class.

#### 3.5.2. Applying the Odds Ratio

Suppose that an Airbnb host is currently listing their <u>shared room</u>. The model predicts the odds that she is a superhost to be 9. If she were instead to list her <u>whole house</u>, what would the model predict the odds of her being a superhost to be?

#### 3.5.3. Room Type and Superhost Status

Visualize the relationships between `host_is_superhost` and `room_type`. 

Does your odds ratio interpretation from 3.5.1 match what your visualization suggests about the relationship between `host_is_superhost` and `room_type`? Explain.

*(Hint: `room_type` should be on your x-axis in this visualization in order to best answer the question above).*